### DISCUSSION TITLE
What data sources would you like instant GraphQL APIs for?

### DISCUSSION THREAD

**Discussion by rajoshighosh** [hasura/graphql-engine#8638] [General]
Today, Hasura gives you instant GraphQL APIs on Postgres (and Postgres family of databases such as Yugabyte, Timescale, Aurora, Citus / Hyperscale), SQL Server and BigQuery. 

We also have a running waitlist for these databases - join the waitlist if you'd like to be notified as soon as we add support: [MongoDB](https://hasura.io/graphql/database/mongodb/),[ MySQL](https://hasura.io/graphql/database/mysql/),[ Oracle](https://hasura.io/graphql/database/oracle/),[ Elastic](https://hasura.io/graphql/database/elasticsearch/), and[ Snowflake](https://hasura.io/graphql/database/snowflake/)

What other data sources are you using that you'd like instant GraphQL APIs on?

**Comment by roamingwheels** [comment]
Let's work on adding a Couchbase connector

**Comment by gamepop** [comment]
Google cloud Spanner

**Reply by sina68** [reply]
google sheet

**Comment by chris-d-edwards** [comment]
Starburst / Trino, I think this could be a variation of the Athena GW connector

**Comment by corysimmons** [comment]
# MySQL so we can use PlanetScale

PlanetScale is VCS for databases. Managing different databases and envs is currently a pain with Hasura. The Hasura CLI just isn't as good as it needs to be.

Postgres > MySQL, but PlanetScale has no intention of ever not-using MySQL, so it's up to Hasura to support PlanetScale via adding support for MySQL.

You guys already have a bunch of work into MySQL (as you have a MySQL Preview and such). Please just dump resources into finishing it up and making it as good as the Postgres implementation.

**Comment by corysimmons** [comment]
# SQLite for _The Edge_

Just saw this https://www.youtube.com/watch?v=yOP5-3_WFus

and this https://twitter.com/dillon_mulroy/status/1545181621606809600

It'd be cool if Hasura supported SQLite (even without every single feature) so we could deploy to the edge and get 50ms db responses from anywhere in the world.

Currently here's how long it takes to query a very simple db from Hasura Console using an admin secret (so not even checking JWT auth or anything). I'm on the USA East Coast and the db is the free Heroku one on the USA West Coast:

![image](https://user-images.githubusercontent.com/224636/178309180-cbef0bfb-fe77-4c8c-9698-68c86635aa23.png)

Sub-sequent requests are only 1s (cold start issue or something???).

**Reply by dvasdekis** [reply]
Including in this the Cloudflare SQLite offering (D1), which is expected to be extremely cost effective relative to standard database hosts

**Comment by fungilation** [comment]
# Tinybird

https://www.tinybird.co/ for realtime analytics apis

**Comment by IsaacLeeWebDev** [comment]
# Apache Cassandra

It seems like it'd be a lot of work to support CQL among other things, but I'd guess that the database is probably the biggest limitation of Hasura's scalability in terms of performance, so support for a more horizontally scalable database would probably be good.

**Comment by javiramos1** [comment]
# MySQL

So we can use [TiDB](https://www.pingcap.com/tidb/).

**Comment by EternalDeiwos** [comment]
# Apache Spark

Because if you can support connecting to a spark context then you can support everything exposed on the backend of that.

**Comment by dameleney** [comment]
Be sure to check out our community call tomorrow where we will be showing an early preview of Hasura support for MySQL and Oracle. 

[Hasura February Community Call](https://hasura.io/community/community-call/)
